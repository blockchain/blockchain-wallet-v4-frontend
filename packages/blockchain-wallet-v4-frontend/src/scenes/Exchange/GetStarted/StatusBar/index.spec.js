import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Status } from './index'

jest.mock('./Statuses/GetStarted', () => 'GetStarted')
jest.mock('./Statuses/InProgress', () => 'InProgress')
jest.mock('./Statuses/Pending', () => 'Pending')
jest.mock('./Statuses/Rejected', () => 'Rejected')
jest.mock('./Statuses/UnderReview', () => 'UnderReview')

describe('Status', () => {
  it('renders correctly when step is getstarted', () => {
    const component = shallow(<Status step='getstarted' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when step is inprogress', () => {
    const component = shallow(<Status step='getstarted' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when step is pending', () => {
    const component = shallow(<Status step='pending' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when step is underreview', () => {
    const component = shallow(<Status step='underreview' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when step is rejected', () => {
    const component = shallow(<Status step='rejected' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
