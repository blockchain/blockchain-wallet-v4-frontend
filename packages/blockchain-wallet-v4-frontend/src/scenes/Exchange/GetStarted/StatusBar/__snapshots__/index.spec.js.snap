// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status renders correctly when step is getstarted 1`] = `
<Fragment>
  <GetStarted />
</Fragment>
`;

exports[`Status renders correctly when step is inprogress 1`] = `
<Fragment>
  <GetStarted />
</Fragment>
`;

exports[`Status renders correctly when step is pending 1`] = `
<Fragment>
  <Pending />
</Fragment>
`;

exports[`Status renders correctly when step is rejected 1`] = `
<Fragment>
  <Rejected />
</Fragment>
`;

exports[`Status renders correctly when step is underreview 1`] = `
<Fragment>
  <UnderReview />
</Fragment>
`;
